src/translations/maps/utils/visuallyIndexed.js

```diff
@@ -0,0 +1,28 @@
+import { arrayFilter } from '../../../helpers/array';
+
+/**
+ * Insert new indexes to the list.
+ *
+ * @private
+ * @param {Array} indexesList List of indexes.
+ * @param {Number} insertionIndex Position inside the actual list.
+ * @param {Array} insertedIndexes List of inserted indexes.
+ * @returns List with new mappings.
+ */
+export function getListWithInsertedItems(indexesList, insertionIndex, insertedIndexes) {
+  return [...indexesList.slice(0, insertionIndex), ...insertedIndexes, ...indexesList.slice(insertionIndex)];
+}
+
+/**
+ * Filter indexes from the list.
+ *
+ * @private
+ * @param {Array} indexesList List of indexes.
+ * @param {Array} removedIndexes List of removed indexes.
+ * @returns Reduced list of mappings.
+ */
+export function getListWithRemovedItems(indexesList, removedIndexes) {
```

Comment:
I find this confusing because it is inconsistent with the names used in `BaseMap`. How about changing this to:

```
/**
 * Filter values from the list.
 *
 * @private
 * @param {Array} list BaseMap's internal data structure
 * @param {Array} removedValues Array of values
 * @returns Filtered list
 */
export function getListWithRemovedValues(list, removedValues) {
```

Reply:
For discussion.